scalafix-testkit/src/main/scala/scalafix/testkit/RuleTest.scala

```diff
@@ -55,17 +56,13 @@ object RuleTest {
         .getOrElse(Conf.Lst(Nil))
       val config = Configuration()
         .withConf(conf)
-        .withScalaVersion(props.scalaVersion)
-        .withScalacOptions(props.scalacOptions)
-        .withScalacClasspath(props.inputClasspath.entries)
+        .withScalaVersion(args.scalaVersion.value)
+        .withScalacOptions(args.scalacOptions)
+        .withScalacClasspath(args.classpath.entries)
       val rules = decoder.read(rulesConf).get.withConfiguration(config).get
       (rules, sdoc)
     }
 
     new RuleTest(test, run)
   }
-
-  def getDialectFromScalaV(scalaV: String): Dialect =
-    if (scalaV.startsWith("3")) ScalafixConfig.Scala3
-    else ScalafixConfig.Scala2
```

Comment:
I tried to centralize dialect handling to make sure the testkit is not drifting. Turning the entire `testsInput` to `-Xsource:3` is a big task, so unfortunately, the testkit codepath is not tested with the new flag.